### ISSUE TITLE
Shield recharge timings make no sense at all

### ISSUE THREAD

**Issue by aeoncleanse** [FAForever/fa#2216]
```    -- Basically run a timer, but with visual bar movement
    ChargingUp = function(self, curProgress, time)
        local owner = self.Owner
        local position = owner:GetPosition()
        local shieldbp = self.Owner:GetBlueprint().Defense.Shield
        local shieldRadius = shieldbp.ShieldSize
        local aiBrain = owner:GetAIBrain()
        local otherShields = aiBrain:GetUnitsAroundPoint((categories.SHIELD * categories.DEFENSE), position, shieldRadius, 'Ally')
        local rechargeTime = time + ((table.getn(otherShields) - 1) * .2 * time)
        if rechargeTime > time * 3 then
            rechargeTime = time
        end
        while curProgress < rechargeTime do
            local fraction = self.Owner:GetResourceConsumed()
            curProgress = curProgress + (fraction / 10)
            curProgress = math.min(curProgress, rechargeTime)

            local workProgress = curProgress / rechargeTime

            self:UpdateShieldRatio(workProgress)
            WaitTicks(1)
        end
    end,```

**Comment by aeoncleanse** [comment]
We increase charge time if there are nearby shields? Since when? We have Shield HP Spillover for that. 
If the time calculated from nearbys is bigger than 3*input, just use input? Wut?

**Comment by MrNukealizer** [comment]
I have to agree. That's a really weird way to go about it. It has been in there since the file was [added to FAF](https://github.com/FAForever/fa/commit/f6feed3615bee3ab172f7700d179d073ca706ccd) but I'm not seeing the logic there. It does explain a lot of weirdness I've seen where one shield can stop artillery but two can't. For reference the original GPG code is: ```ChargingUp = function(self, curProgress, time)
        while curProgress < time do
            local fraction = self.Owner:GetResourceConsumed()
            curProgress = curProgress + ( fraction / 10 )
            curProgress = math.min( curProgress, time )
            
            local workProgress = curProgress / time
            
            self:UpdateShieldRatio( workProgress )
            WaitTicks(1)
        end    
    end,```

**Comment by CookieNoob** [comment]
my guess is that this is the first attempt to nerf overlapping shields before damage transfer was introduced. After that was added, no one removed that code.

**Comment by aeoncleanse** [comment]
I agree. I'm going to move for removing this to simplify shield mechanics for players, maybe publish a bullet point guide of how shields work afterwards.

**Comment by JaggedAppliance** [comment]
Can someone explain how large the effect is in-game?

**Comment by MrNukealizer** [comment]
If two shields are in range of each other they take 20% longer to recharge, three shields 40% longer, 4 shields 60% longer, etc. It tops out at 10 shields in range of each other, at which point they take three times as long to recharge. Once you add an 11th shield the penalty goes away and they're all back to normal.
